stack.py

```diff
@@ -8,10 +8,10 @@ def push(self, value):
         self._storage.push(value)
 
     def pop(self):
-        return self._storage.pop()
-
-    def size(self):
-        return len(self._storage)
+        try:
+            return self._storage.pop()
+        except ValueError as exception:
+            raise ValueError("Cannot pop from an empty stack") from exception
```

Comment:
Typically this would be a great way to re-raise an exception because it maintains the stack trace, which can help one figure out what actually went wrong. But in this case nothing really went wrong, it is expected behavior for the Stack to ValueError when it's called empty.  We don't need to know and shouldn't surface to our end users that it originated from a ValueError of a linked list. 
```suggestion
        except ValueError:
            raise ValueError("Cannot pop from an empty stack")
```